compiler/build.gradle

```diff
@@ -54,47 +54,68 @@ def addLibraryIfNotLinked = { libName, argList ->
 }
 
 model {
-  toolChains {
-    gcc(Gcc) {
-      target("x86_64") {
-        cppCompiler.withArguments { args ->
-          args << "-m64"
-        }
-        linker.withArguments { args ->
-          args << "-m64"
-        }
+  if (System.getProperty('use.vc') != null) {
+    toolChains {
+      visualCpp(VisualCpp) {
       }
-      target("x86_32") {
-        cppCompiler.withArguments { args ->
-          args << "-m32"
-        }
-        linker.withArguments { args ->
-          args << "-m32"
-        }
+    }
+    platforms {
+      // VisualCpp supports 'x86' and 'amd64', but it builds 'x86' by default.
+      // Thus Protobuf is likely to have been built for x86. No matter what the
+      // detected architecture is, 'x86' is the safest target.
+      x86_32 {
+        architecture 'x86'
+      }
+      x86_64 {
+        architecture 'x86'
```

Comment:
To build 64-bit codegen, you need to have Protobuf compiled in 64-bit in VC, which I don't know whether people would want to take the effort to do. It's more likely that 64-bit JVM is installed in the system, thus osdetector would report 'x86_64', but Protobuf has been compiled in 32-bit, and you would want to compile 32-bit codegen too.